### PR TITLE
ft: update versioning doc to reflect changes

### PR DESCRIPTION
**Pull request by electrachong**
Will need @nicolas2bert's help to figure out how to render the documents for preview on the readthedocs :smile:

See rendered changes here: http://s3-server.readthedocs.io/en/doc-adddoc/ARCHITECTURE/#versioning

### CODE REVIEW

File: docs/ARCHITECTURE.rst [hunk 1]
```diff
@@ -368,7 +368,11 @@ version. S3 checks if versioning is suspended or has never been
 configured, and sets the ``versionId`` option to ``''`` in PUT calls to
 the metadata engine when creating a new null version.
 
-The tables below show the keys resulting from PUT calls to metadata if
+If the master version is a null version, S3 also sends a DELETE call to metadata
+prior to the PUT, in order to clean up any pre-existing null versions which may,
+in certain edge cases, have been stored as a separate version.[#1]_
```
Comment (bennettbuchanan) [inline docs/ARCHITECTURE.rst]: This footnote doesn't seem to be rendering properly when previewed.

**Comment by jonathan-gramain** [comment]
When https://github.com/scality/Arsenal/pull/262 gets merged, it will change a bit how versioning code behaves:

The "Metadata Versioning Options" should be updated to reflect this: when `versioning` and `versionId` are both set, instead of saying that it returns an error, it could say that it's only used for internal needs (e.g. CRR). What is does is create or update a specific version + the master only if the master is older than the written version. I'm not convinced we need to explain this in the doc though.

I guess the easiest is to update the doc in this PR directly.

**Comment by electrachong** [comment]
@jonathan-gramain does the update look OK for when your pr is merged?

File: docs/ARCHITECTURE.rst [hunk 2]
```diff
@@ -192,10 +192,9 @@ PUT
       versions once versioning has been suspended, which is discussed in
       `"Null Version Management" <#null-version-management>`__.
 
-Only one option is used at a time. ``versionId: <versionId>`` does not
-have to be used with ``versioning: true`` set to work, nor should they
-both be set. If both are used at once, the metadata engine will return
-an error.
+In general, only one option is used at a time. When ``versionId`` and
+``versioning`` are both set, it is for internal needs (e.g. cross-region
```
Comment (LaurenSpiegel) [inline docs/ARCHITECTURE.rst]: If versionId and versioning is set, then a new version with versionId will be created and master will be updated if the versionId is later than the current master.

**Comment by electrachong** [comment]
@jonathan-gramain updated again - if you can proof sometime, can you confirm you agree with the description?

**Comment by jonathan-gramain** [comment]
@electrachong Looks good to me!

**Pull request merged**